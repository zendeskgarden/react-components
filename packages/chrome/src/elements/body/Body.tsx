/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { HTMLAttributes } from 'react';
import PropTypes from 'prop-types';
import { StyledBody } from '../../styled';
import { BodyContext } from '../../utils/useBodyContext';

interface IBodyProps {
  /** Adjusts the body content height to allow space for a footer component */
  hasFooter?: boolean;
}

/**
 * @extends HTMLAttributes<HTMLDivElement>
 */
export const Body = React.forwardRef<HTMLDivElement, IBodyProps & HTMLAttributes<HTMLDivElement>>(
  ({ hasFooter, ...props }, ref) => {
    const bodyContextValue = { hasFooter: !!hasFooter };

    return (
      <BodyContext.Provider value={bodyContextValue}>
        <StyledBody ref={ref} {...props} />
      </BodyContext.Provider>
    );
  }
);

Body.displayName = 'Body';

Body.propTypes = {
  hasFooter: PropTypes.bool
};
