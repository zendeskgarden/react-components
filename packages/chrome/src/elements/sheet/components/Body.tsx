/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { forwardRef, HTMLAttributes } from 'react';
import { StyledSheetBody } from '../../../styled';

const SheetBody = forwardRef<HTMLDivElement, HTMLAttributes<HTMLDivElement>>((props, ref) => {
  return <StyledSheetBody ref={ref} {...props} />;
});

SheetBody.displayName = 'Sheet.Body';

/**
 * @extends HTMLAttributes<HTMLDivElement>
 */
export const Body = SheetBody;
