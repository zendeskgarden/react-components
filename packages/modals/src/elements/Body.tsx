/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { HTMLAttributes } from 'react';
import { StyledBody } from '../styled';
import { useModalContext } from '../utils/useModalContext';

/**
 * @extends HTMLAttributes<HTMLDivElement>
 */
export const Body = React.forwardRef<HTMLDivElement, HTMLAttributes<HTMLDivElement>>(
  (props, ref) => {
    const { getContentProps } = useModalContext();

    return <StyledBody ref={ref} {...getContentProps(props)} />;
  }
);

Body.displayName = 'Body';
