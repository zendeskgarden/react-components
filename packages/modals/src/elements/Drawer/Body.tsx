/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { HTMLAttributes, forwardRef } from 'react';
import { useModalContext } from '../../utils/useModalContext';
import { StyledDrawerBody } from '../../styled';

const BodyComponent = forwardRef<HTMLDivElement, HTMLAttributes<HTMLDivElement>>((props, ref) => {
  const { getContentProps } = useModalContext();

  return (
    <StyledDrawerBody {...(getContentProps(props) as HTMLAttributes<HTMLDivElement>)} ref={ref}>
      {props.children}
    </StyledDrawerBody>
  );
});

BodyComponent.displayName = 'Drawer.Body';

/**
 * @extends HTMLAttributes<HTMLDivElement>
 */
export const Body = BodyComponent;
