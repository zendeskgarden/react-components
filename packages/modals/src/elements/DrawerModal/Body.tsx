/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React from 'react';
import { useModalContext } from '../../utils/useModalContext';
import { StyledDrawerModalBody } from '../../styled';

/**
 * @extends HTMLAttributes<HTMLDivElement>
 */
export const Body = React.forwardRef<HTMLDivElement, React.HTMLAttributes<HTMLDivElement>>(
  (props, ref) => {
    const { getContentProps } = useModalContext();

    return (
      <StyledDrawerModalBody ref={ref} {...getContentProps(props)}>
        {props.children}
      </StyledDrawerModalBody>
    );
  }
);

Body.displayName = 'DrawerModal.Body';
