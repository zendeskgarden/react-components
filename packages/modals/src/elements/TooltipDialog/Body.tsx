/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { HTMLAttributes, forwardRef } from 'react';
import { StyledTooltipDialogBody } from '../../styled';
import { useTooltipDialogContext } from '../../utils/useTooltipDialogContext';

const BodyComponent = forwardRef<HTMLDivElement, HTMLAttributes<HTMLDivElement>>((props, ref) => {
  const { getContentProps } = useTooltipDialogContext();

  return (
    <StyledTooltipDialogBody
      {...(getContentProps(props) as HTMLAttributes<HTMLDivElement>)}
      ref={ref}
    />
  );
});

BodyComponent.displayName = 'TooltipDialog.Body';

/**
 * @extends HTMLAttributes<HTMLDivElement>
 */
export const Body = BodyComponent;
