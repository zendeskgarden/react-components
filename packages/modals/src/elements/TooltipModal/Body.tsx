/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { HTMLAttributes } from 'react';
import { StyledTooltipModalBody } from '../../styled';
import { useTooltipModalContext } from '../../utils/useTooltipModalContext';

/**
 * @extends HTMLAttributes<HTMLDivElement>
 */
export const Body = React.forwardRef<HTMLDivElement, HTMLAttributes<HTMLDivElement>>(
  (props, ref) => {
    const { getContentProps } = useTooltipModalContext();

    return <StyledTooltipModalBody ref={ref} {...getContentProps(props)} />;
  }
);

Body.displayName = 'TooltipModal.Body';
