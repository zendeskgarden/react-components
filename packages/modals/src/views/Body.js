import styled from 'styled-components';
import { retrieveTheme } from '@zendesk/garden-react-theming';
import ModalStyles from '@zendesk/garden-css-modals';

import { version } from '../../package.json';
const COMPONENT_ID = 'modals.body';

/**
 * Accepts all `<div>` props
 */
const Body = styled.div.attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': version,
  className: ModalStyles['c-dialog__body']
})`
  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

/** @component */
export default Body;
