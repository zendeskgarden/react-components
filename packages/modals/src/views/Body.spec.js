import React from 'react';
import { shallow } from 'enzyme';
import Body from './Body';

describe('Body', () => {
  it('renders default styling', () => {
    const wrapper = shallow(<Body />);

    expect(wrapper).toMatchSnapshot();
  });
});
