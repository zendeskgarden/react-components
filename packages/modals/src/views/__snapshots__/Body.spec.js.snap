// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body renders default styling 1`] = `
<div
  className="c-dialog__body "
  data-garden-id="modals.body"
  data-garden-version="version"
/>
`;
